lib/BIGSdb/Plugins/PhyloViz.pm

```diff
@@ -0,0 +1,415 @@
+	    return;
+	} else {
+	    # From here, with parameters retrieved, we need to build the 2 files needed for PhyloViz:
+	    # - Profile data
+	    # - Auxiliary data
+	    $| =  1;
+	    say q(<div class="box" id="resultstable">);
+	    say q(<p>Please wait for processing to finish (do not refresh page).</p>);
+	    say q(<p class="hideonload"><span class="main_icon fa fa-refresh fa-spin fa-4x"></span></p>);
+	    say q(<p>Data are being processed and sent to PhyloViz Online.</p>);
+
+	    my $uuid = BIGSdb::Utils::get_random();
+	    my $profile_file = join("/", $self->{'config'}->{'tmp_dir'}, "profile_data_$uuid.txt");
+	    my $auxiliary_file = join("/", $self->{'config'}->{'tmp_dir'}, "auxiliary_data_$uuid.txt");
+	    if( $self->generate_profile_file({'file' => $profile_file, 'schemes' => $selected_schemes,
+					      'isolates' => $isolates_ids, 'loci' => $selected_loci})){
+		say q(<div class="box" id="statusbad"><p>Nothing found in the database for your isolates!</p></div>);
+		return;
+	    }
+	    $self->generate_auxiliary_file({'file' => $auxiliary_file, 'schemes' => $selected_schemes,
+					    'isolates' => $isolates_ids, 'fields' => $selected_isolates_fields});
+	    # Upload data files to phyloviz online using python script
+	    my ($phylo_id, $msg) = $self->upload_data_to_phyloviz({'profile'   => $profile_file,
+								   'auxiliary' => $auxiliary_file});
+	    if( !$phylo_id ){
+		say "\n<div class=\"box\" id=\"statusbad\"><p>Something went wrong: $msg</p></div>";
+		return;
+	    }
+
+	    say qq(<p>Click this <a href="$phylo_id" target="_blank">link</a> to view your tree</p>);
+	    say q(</div>);
+	    return;
+	}
+    }
+    say q[<div class="box" id="queryform"><p>PhyloViz: This plugin allows the analysis of sequence-based ]
+	. q[typing methods that generate allelic profiles and their associated epidemiological data.</p>   ];
+    say $q->start_form();
+    # Selected isolates
+    if( $self->{'config'}->{'phyloviz_show_isolates_ids'} eq 'yes' ){
+	$self->print_selected_isolates({'selected_ids' => $isolates_ids, 'size' => 11});
+    }
+    # Isolates fields
+    $self->print_isolates_fieldset(1);
+    # Loci fieldset
+    $self->print_isolates_locus_fieldset();
+    # Schemes Tree
+    $self->print_scheme_fieldset({'fields_or_loci' => 0});
+    # Action button (Submit only due to 'no_reset => 1')
+    $self->print_action_fieldset({ no_reset => 1});
+    say $q->hidden($_) foreach qw (db page name query_file scheme_id set_id list_file datatype);
+    say $q->end_form();
+    say q(</div>);
+    return;
+}
+
+sub print_selected_isolates {
+
+    my ($self, $options) = @_;
+    if ( ! scalar($options->{'selected_ids'}) ){
+	say q(No isolates selected found);
+    }
+    else {
+	# Always search in table isolates, not the VIEW allowing to use this plugin with freshly created isolates
+	my $query = "SELECT isolates.id, isolates.isolate FROM isolates WHERE isolates.id IN (";
```

Comment:
The isolate field is not guaranteed to be present in the isolates table.  You should use $self->{'system'}->{'view'}.$self->{'system'}->{'labelfield'} since you can substitute isolates for another field (e.g. strain) using the labelfield system tag.